platform/commonUI/edit/src/creation/LocatorController.js

```diff
@@ -81,11 +81,64 @@ define(
                         !!$scope.treeModel.selectedObject
                     );
                 }
+
+                // Check if create new folder is a valid action for selected object
+                $scope.validParent = function () {
+                    if ($scope.treeModel.selectedObject) {
```

Comment:
It will be better if you can cache the $scope.treeModel in a variable as it has been used at multiple places here.